Core/include/Acts/TrackFinding/TrackSelector.hpp

```diff
@@ -350,14 +357,19 @@ inline bool TrackSelector::EtaBinnedConfig::hasCuts(double eta) const {
 }
 
 inline std::size_t TrackSelector::EtaBinnedConfig::binIndex(double eta) const {
-  if (!hasCuts(eta)) {
+  std::size_t index = binIndexNoCheck(eta);
+  if (!(index + 1 < absEtaEdges.size())) {
     throw std::invalid_argument{"Eta is outside the abs eta bin edges"};
   }
+  return index;
+}
 
+inline std::size_t TrackSelector::EtaBinnedConfig::binIndexNoCheck(double eta) const {
   auto binIt =
       std::upper_bound(absEtaEdges.begin(), absEtaEdges.end(), std::abs(eta));
-  std::size_t index = std::distance(absEtaEdges.begin(), binIt) - 1;
-  return index;
+  std::size_t index = std::distance(absEtaEdges.begin(), binIt);
+  if (index == 0) index = absEtaEdges.size() + 1; // positive value to check for underflow
+  return index - 1;
```

Comment:
Since I was adding some other improvements to this file, I also added my long-wished for `binIndexNoCheck()`. This saves a `double` comparison and will eventually simplify client code.

It also fixes a bug where if `eta < absEtaEdges[0]` it would return index `-1` (ie. max `size_t`) and give an array out of bounds exception. Likely never a problem because any sensible client will have `absEtaEdges[0] = 0`.